packages/core/src/new-api/internal/new-execution/types/execution-result.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  FailedEvmExecutionResult,
+  SuccessfulEvmExecutionResult,
+} from "./evm-execution";
+
+/**
```

Comment:
I used to have an extra type: `SIMULATION_ERROR`.

I removed it as those errors shouldn't be considered permanent (i.e. not journaled), because a re-run may make them pass.

I think I regret having removed it. It creates this new entity in the system, where execution may result in an ExecutionResult or a simulation error. 

I'm starting to think that I should reintroduce it, and we can decide not to journal it anyways.

WDYT?